packages/plugin-js-packages/src/lib/runner/audit/unify-type.ts

```diff
@@ -189,11 +191,66 @@ export function yarnv2ToAuditResult(output: string): AuditResult {
     },
   );
 
-  const total = Object.values(yarnv2Audit.metadata.vulnerabilities).reduce(
-    (acc, value) => acc + value,
-    0,
+  return {
+    vulnerabilities,
+    summary: {
+      ...yarnv2Audit.metadata.vulnerabilities,
+      total: getVulnerabilitiesTotal(yarnv2Audit.metadata.vulnerabilities),
+    },
+  };
+}
+
+export function pnpmToAuditResult(output: string): AuditResult {
+  const pnpmResult = JSON.parse(output) as PnpmAuditResultJson;
+
+  const vulnerabilities = Object.values(pnpmResult.advisories).map(
+    ({
+      module_name: name,
+      id,
+      title,
+      url,
+      severity,
+      vulnerable_versions: versionRange,
+      recommendation: fixInformation,
+      findings,
+    }): Vulnerability => {
+      const path = findings[0]?.paths[0];
+
+      return {
+        name,
+        id,
+        title,
+        url,
+        severity,
+        versionRange,
+        directDependency: path == null ? true : pnpmToDirectDependency(path),
+        fixInformation,
+      };
+    },
   );
-  const summary = { ...yarnv2Audit.metadata.vulnerabilities, total };
 
-  return { vulnerabilities, summary };
+  return {
+    vulnerabilities,
+    summary: {
+      ...pnpmResult.metadata.vulnerabilities,
+      total: getVulnerabilitiesTotal(pnpmResult.metadata.vulnerabilities),
+    },
+  };
+}
+
+export function pnpmToDirectDependency(path: string): string | true {
+  // the format is ". > <direct dependency>@<version> > ... > <current dependency>@<version>"
+  const deps = path.split(' ').filter(item => item !== '>' && item !== '.');
```

Comment:
Since ` > ` is the separator, this would be more intuitive:

```suggestion
  const deps = path.split(' > ').filter(item => item !== '.');
```

If `.` is always at the start, could also be:

```suggestion
  const deps = path.split(' > ').slice(1);
```

Reply:
Good point!